src/System.Private.Uri/src/System/UriBuilder.cs

```diff
@@ -276,7 +276,7 @@ public string Query
                 }
                 if (value.Length > 0)
                 {
-                    value = '?' + value;
+                    value = value.StartsWith("?") ? value : '?' + value;
```

Comment:
Assuming this is a fix that's desired, not a breaking change we care about, etc. I'd think it'd be a bit better to write it as:
```C#
if (value.Length > 0 && value[0] != '?')
{
    value = "?" + value;
}
```